src/Object/MetadataInterface.php

```diff
@@ -13,10 +13,21 @@
 
 namespace Sonata\AdminBundle\Object;
 
-/**
- * NEXT_MAJOR: Remove CoreBundle dependency.
- */
-interface MetadataInterface extends \Sonata\CoreBundle\Model\MetadataInterface
+use Sonata\BlockBundle\Meta\MetadataInterface as BlockMetadataInterface;
+use Sonata\CoreBundle\Model\MetadataInterface as CoreMetadataInterface;
+
+// NEXT_MAJOR: Remove CoreBundle dependency.
+if (interface_exists(CoreMetadataInterface::class)) {
+    interface InternalMetadataInterface extends CoreMetadataInterface, BlockMetadataInterface
+    {
+    }
+} else {
+    interface InternalMetadataInterface extends BlockMetadataInterface
```

Comment:
Same metadata, does it have to do with BlockBundle, can we remove this extends?

Reply:
https://github.com/sonata-project/SonataMediaBundle/pull/1744#discussion_r437643302